.github/workflows/ci.yaml

```diff
@@ -77,21 +78,29 @@ jobs:
 
       - name: Regenerate IDL files for Fast-DDS and Fast-DDS-python
         run: |
-          cd src/Fast-DDS/utils/scripts
-          ./update_generated_code_from_idl.sh
+          cd src/Fast-DDS
+          ./utils/scripts/update_generated_code_from_idl.sh
           cd -
           cd src/Fast-DDS-python/fastdds_python/test/types
           ./generate.sh
           cd -
           cd src/Fast-DDS-python/fastdds_python_examples/HelloWorldExample
-          fastddsgen -python -replace HelloWorld.idl
+          fastddsgen -cdr both -python -replace HelloWorld.idl
 
       - name: Build workspace
         run: |
           cmake --version
           colcon build --event-handlers=console_direct+ --mixin rel-with-deb-info ccache
 
-      - name: Test fastddsgen
+      - name: Test fastddsgen with fastcdr v1
+        if: startsWith(matrix.fastcdr_version, 'v1')
+        run: |
+          source install/local_setup.bash
+          cd src/fastddsgen
+          ./gradlew test -Dcdr_version=v1 -Dblacklist_tests=external,maps,optional
```

Comment:
Personally, I think that in order to ensure that Fast DDS-Gen is compatible with Fast CDR v1, every IDL file must pass successfully the test suite.